src/uiRouterNgModule.ts

```diff
@@ -16,12 +16,30 @@ import { ROUTES } from "@angular/router";
 /** @hidden */ export const UIROUTER_STATES       = new OpaqueToken("UIRouter States");
 // /** @hidden */ export const ROUTES = UIROUTER_STATES;
 
+export function onTransitionReady(transitionService: TransitionService, plateformId) {
+    if (isPlatformServer(plateformId)) {
+       return () => Promise.resolve();
+    }
+
+    return () => new Promise(resolve => {
```

Comment:
update into:
```ts
    return () => new Promise(resolve => {
      const hook = trans => { trans.promise.then(resolve, resolve); };
      transitionService.onBefore({}, hook);
    });
```
the circular deps is an issue on my side, `router.globals.start$` should work too but I prefer this one as it's more simple